tests/csapi/power_levels_test.go

```diff
@@ -40,3 +43,83 @@ func TestDemotingUsersViaUsersDefault(t *testing.T) {
 		},
 	})
 }
+
+func TestPowerLevels(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	roomID := alice.CreateRoom(t, map[string]interface{}{})
+
+	// sytest: GET /rooms/:room_id/state/m.room.power_levels can fetch levels
+	t.Run("GET /rooms/:room_id/state/m.room.power_levels can fetch levels", func(t *testing.T) {
+		// Test if the old state still exists
+		res := alice.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "rooms", roomID, "state", "m.room.power_levels"})
+
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 200,
+			JSON: []match.JSON{
+				match.JSONKeyPresent("users"),
+			},
+		})
+	})
+
+	// sytest: PUT /rooms/:room_id/state/m.room.power_levels can set levels
+	t.Run("PUT /rooms/:room_id/state/m.room.power_levels can set levels", func(t *testing.T) {
```

Comment:
This should be setting the PL for another user. It's very likely alice is already PL 100.
Further you should check that the event comes down sync with the PL you expect?

Further, is this even calling the endpoint mentioned in the test name? (I'm not sure whether it's using a more generic endpoint for sending events — possibly it's fine, I just don't find it as obvious as SyTest :eyes: .)

https://github.com/matrix-org/sytest/blob/fd4f4132913e9575471af515f67c633d2e3123d3/tests/10apidoc/36room-levels.pl#L38-L72

Reply:
> Further, is this even calling the endpoint mentioned in the test name? (I'm not sure whether it's using a more generic endpoint for sending events — possibly it's fine, I just don't find it as obvious as SyTest eyes .)

On this: I'd be tempted to call the endpoint manually (rather than relying on `SendEventSynced` doing it for you), personally, but maybe worth checking with a Complement expert.

Reply:
> On this: I'd be tempted to call the endpoint manually

It calls `/state/m.room.power_levels`, which `SendEventSynced` calls (when it sees its a state event), so its the same endpoint.

Otherwise good points for improvement 👍